src/Features/Lsif/GeneratorTest/FoldingRangeTests.vb

```diff
@@ -0,0 +1,115 @@
+{
+}|}", Nothing)>
+        <InlineData("
+class C{|foldingRange:
+{
+    void M(){|foldingRange:
+    {
+        for (int i = 0; i < 10; i++){|foldingRange:
+        {
+            M();
+        }|}
+    }|}
+}|}", Nothing)>
+        <InlineData("
+{|foldingRange:#region
+#endregion|}", Nothing)>
+        <InlineData("
+using {|foldingRange:System;
+using System.Linq;|}", FoldingRangeKind.Imports)>
+        <InlineData("
+using {|foldingRange:S = System.String; 
+using System.Linq;|}", FoldingRangeKind.Imports)>
+        <InlineData("
+{|foldingRange:// Comment Line 1
+// Comment Line 2|}", Nothing)>
+        Public Async Function TestFoldingRanges(code As String, rangeKind As FoldingRangeKind?) As Task
+            Using workspace = TestWorkspace.CreateWorkspace(
+                    <Workspace>
+                        <Project Language="C#" AssemblyName=<%= TestProjectAssemblyName %> FilePath="Z:\TestProject.csproj" CommonReferences="true">
+                            <Document Name="A.cs" FilePath="Z:\A.cs">
+                                <%= code %>
+                            </Document>
+                        </Project>
+                    </Workspace>)
+
+                Dim annotatedLocations = AbstractLanguageServerProtocolTests.GetAnnotatedLocations(workspace, workspace.CurrentSolution)
+                Dim expectedRanges = annotatedLocations("foldingRange").Select(Function(location) CreateFoldingRange(rangeKind, location.Range)).ToArray()
+
+                Dim document = workspace.CurrentSolution.Projects.Single().Documents.Single()
+                Dim lsif = Await TestLsifOutput.GenerateForWorkspaceAsync(workspace)
+                Dim actualRanges = lsif.GetFoldingRanges(document)
+
+                AssertEx.SetEqual(expectedRanges, actualRanges, FoldingRangeComparer.Instance, itemInspector:=AddressOf GetStringForAssert)
+            End Using
+        End Function
+
+        Private Shared Function CreateFoldingRange(kind As FoldingRangeKind?, range As Range) As FoldingRange
+            Return New FoldingRange() With
+            {
+                .Kind = kind,
+                .StartCharacter = range.Start.Character,
+                .EndCharacter = range.End.Character,
+                .StartLine = range.Start.Line,
+                .EndLine = range.End.Line
+            }
+        End Function
+
+        Private Shared Function GetStringForAssert(range As FoldingRange) As String
+            Dim rangeKindStr = $"Kind: {If(range.Kind?.ToString(), "null")}"
+            Dim startLineStr = $"StartLine: {range.StartLine}"
+            Dim endLineStr = $"EndLine: {range.EndLine}"
+            Dim startCharStr = $"StartCharacter: {range.StartCharacter}"
+            Dim endCharStr = $"EndCharacter: {range.EndCharacter}"
+            Return $"[{String.Join(", ", rangeKindStr, startLineStr, endLineStr, startCharStr, endCharStr)}]"
+        End Function
+
+        Private NotInheritable Class FoldingRangeComparer
```

Comment:
Did we not have this in the core LSP tests to do the same thing?

Reply:
Thanks - it did not initially work for me as there were multiple ranges, but I had to just fix the test. Was able to get rid of this comparer and use the same test helper as in LSP tests.